linkerd/stack/src/gate.rs

```diff
@@ -72,7 +72,7 @@ impl Rx {
     }
 
     /// Waits for the gate state to change to be open.
-    async fn acquire(&mut self) -> Option<OwnedSemaphorePermit> {
+    pub async fn acquire(&mut self) -> Option<OwnedSemaphorePermit> {
```

Comment:
It's a shame this needs to be `pub` just for tests. Do we have a `test-util` feature in this crate already? It would be ideal to expose this only as a test-only utility (e.g. `pub async fn acquire_for_test` which wraps non-pub `async fun acquire`).

Reply:
Can do.